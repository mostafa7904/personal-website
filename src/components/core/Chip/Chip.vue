<template>
  <div class="chip">
    <div class="chip__content">
      <slot></slot>
    </div>
    <div class="chip__gradient"></div>
  </div>
</template>

<style lang="scss">
.chip {
  position: relative;
  display: flex;
  align-items: center;
  max-height: 20px;

  &__gradient {
    position: absolute;
    left: 0%;
    top: 0%;
    right: 0%;
    bottom: 0%;
    background-image: linear-gradient(
      270deg,
      rgba(255, 56, 104, 0.5),
      rgba(77, 84, 255, 0.5)
    );
    filter: blur(6px);
  }
  &__content {
    position: relative;
    z-index: 10;
    background-color: #13111a;
    border-style: solid;
    border-width: 1px;
    border-color: rgba(255, 255, 255, 0.07);
    padding: 7px 20px;
    border-radius: 12px;
    font-size: 0.6rem;
    font-weight: 700;
    letter-spacing: 0;
  }
}
</style>
