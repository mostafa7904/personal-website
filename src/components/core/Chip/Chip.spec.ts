import { describe, expect, it } from 'vitest'
import { type MountingOptions, mount } from '@vue/test-utils'
import Chip from './Chip.vue'

const mountFunction = (options?: MountingOptions<any>) => {
  return mount(Chip, {
    ...options,
  })
}

describe('Chip.vue', () => {
  it('should be truthy', () => {
    const wrapper = mountFunction()
    expect(wrapper.vm).toBeTruthy()
  })
  it('should render correctly', () => {
    const wrapper = mountFunction({
      slots: {
        default: 'test',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
