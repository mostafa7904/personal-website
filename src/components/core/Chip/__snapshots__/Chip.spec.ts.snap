// Vitest Snapshot v1

exports[`Chip.vue > should render correctly 1`] = `
"<div class=\\"chip\\">
  <div class=\\"chip__content\\">test</div>
  <div class=\\"chip__gradient\\"></div>
</div>"
`;
